contracts/GenericBondCalculator.sol

```diff
@@ -36,16 +35,23 @@ contract GenericBondCalculator {
         tokenomics = _tokenomics;
     }
 
-    /// @dev Calculates the amount of OLAS tokens based on the bonding calculator mechanism.
-    /// @notice Currently there is only one implementation of a bond calculation mechanism based on the UniswapV2 LP.
-    /// @notice IDF has a 10^18 multiplier and priceLP has the same as well, so the result must be divided by 10^36.
+    /// @dev Calculated inverse discount factor.
+    /// @return idf Inverse discount factor in 18 decimals format.
+    function calculateIDF(bytes memory) public view virtual returns (uint256 idf) {
+        // Note: IDF is deprecated in Tokenomics, and can be assumed as equal to 1e18 by default
+        idf = 1e18;
+    }
+
+    /// @dev Calculates the amount of OLAS tokens based on the bonding calculator mechanism accounting for dynamic IDF.
     /// @param tokenAmount LP token amount.
     /// @param priceLP LP token price.
+    /// @param data Custom data to calculate the IDF.
     /// @return amountOLAS Resulting amount of OLAS tokens.
-    /// #if_succeeds {:msg "LP price limit"} priceLP * tokenAmount <= type(uint192).max;
-    function calculatePayoutOLAS(uint256 tokenAmount, uint256 priceLP) external view
-        returns (uint256 amountOLAS)
-    {
+    function calculatePayoutOLAS(
+        uint256 tokenAmount,
+        uint256 priceLP,
+        bytes memory data
```

Comment:
I'm not sure that we should use implicit pack of arguments as a raw sequence of bytes (memory data) and then unpack it (abi.decode) instead of explicitly enumerating the arguments.